### PR TITLE
AttendenceRegisterExtended contract added (issue #38)

### LINKED ISSUE #38: AttendanceRegisterExtended
Standard: 3
Sequence: 2
Contract Name: AttendanceRegisterExtended
Problem Statement: https://github.com/Aniket-Engg/solidity-school/tree/master/std-3/2_AttendanceRegisterExtended

Solve the problem statement.

Contribution guidelines: https://github.com/Aniket-Engg/solidity-school/wiki/Contribution-Guidelines

### PR DESCRIPTION
**Pull request by RijuChakraborty**
fixes #38 

### New file added-
AttendenceRegisterExtended contract added which is inheriting from AttendenceRegister and includes a function which takes roll number as input and returns the details of the corresponding student.

### Enhancements-
Removed "rollNumber = 0" from constructor in AttendenceRegister contract as solidity initialize uint variables with 0 by default.
This change also helped in reducing gas costs by ~3k units.

@Aniket-Engg please have a look.

### CODE REVIEW

File: std-3/2_AttendanceRegisterExtended/AttendanceRegisterExtended.sol [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.17;
+
+import "../../std-2/5_AttendanceRegister/AttendanceRegister.sol";
+
+contract AttendanceRegisterExtended is AttendanceRegister{
+
+    function getStudentByRoll (uint256 _roll) public view returns (Student memory){
```
Comment (Aniket-Engg) [inline std-3/2_AttendanceRegisterExtended/AttendanceRegisterExtended.sol:9]: Please consider adding NatSpec

Comment (RijuChakraborty) [reply std-3/2_AttendanceRegisterExtended/AttendanceRegisterExtended.sol:15]: Done. Please have a look

**Review by Aniket-Engg — APPROVED** [review]

**Comment by Aniket-Engg** [comment]
Thanks @RijuChakraborty !!!

**Comment by RijuChakraborty** [comment]
> Thanks @RijuChakraborty !!!

Welcome :)

**Pull request merged**